### DISCUSSION TITLE
Change Path to pages (Deployment to multiple subdomains)

### DISCUSSION THREAD

**Discussion by BenAllenUK** [vercel/next.js#22718] [Ideas]
### Describe the feature you'd like to request

I would like the ability to change the path to the pages folder via the config file and CLI. This will allow me to build, run and deploy multiple apps at the same time from one folder. 

My use case is that I can have multiple "pages" that represent subdomains:
- `src/_root_/pages` -> `example.com`
- `src/docs/pages` -> `docs.example.com`
- `src/admin/pages`-> `admin.example.com`

Having a configurable path means I can build and deploy multiple apps at the same time. At the moment I must move the contents of the file from `src/_pages/[docs|_root_|admin]` to `src/pages` prior to deploying, building or running locally. For a CI/CD deployment, I have to clone the repo into multiple folder locations or across multiple machines to deploy so that the deployments avoid overriding other deployments.

### Describe the solution you'd like

When I run `next dev`, I can specify the path to the pages i.e `--pages-dir=src/admin/pages`. This will allow me to have multiple instances of the app running from one folder when also specifying a different port.

When I deploy, I can then run the deployment command with `--pages-dir=src/admin/pages` and a unique build directory, allowing me to build and deploy from one repo multiple apps to multiple subdomains. 

I may also want this configurable via `next.config.js`.

### Describe alternatives you've considered

Treating `src/pages` as a special folder and bash scripting a move of files into that folder prior to deployment/build/development running.

**Comment by KevinDanikowski** [comment]
I too have the same issue, use case being multiple subdomains in the same repo. Another solution would simply be just specify the `_app.js` and `_document.js` location:

next.config.js

` module.exports = { appJsPath: './src/pages/subdomain/_app.js' }` for example
